### ISSUE TITLE
"stg mail -a -e" fails with a TypeError exception

### ISSUE THREAD

**Issue by chucklever** [ctmarinas/stgit#58]
When using the "-a" and "-e" options together on "stg mail", an exception occurs. Using either alone, or specifying all the patches in the series using a range expression (ie, xxx..yyy) does not cause a problem.

```
[cel@klimt linux]$ stg mail -e -a -vv1 --to=nfs --to=rdma
Checking the validity of the patches ... done
Error: Unhandled exception:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/stgit/main.py", line 193, in _main
    ret = command.func(parser, options, args)
  File "/usr/lib/python2.7/site-packages/stgit/commands/mail.py", line 816, in func
    msg_id = __send_message('cover', tmpl, options, patches)
  File "/usr/lib/python2.7/site-packages/stgit/commands/mail.py", line 368, in __send_message
    'patch': (__build_message, 'patch "%s"' % args[0]),
TypeError: not all arguments converted during string formatting
[cel@klimt linux]$ stg mail -e -a -vv1 --to=nfs --to=rdma
Checking the validity of the patches ... done
Error: Unhandled exception:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/stgit/main.py", line 195, in _main
    ret = command.func(parser, options, args)
  File "/usr/lib/python2.7/site-packages/stgit/commands/mail.py", line 877, in func
    msg_id = __send_message('cover', tmpl, options, patches)
  File "/usr/lib/python2.7/site-packages/stgit/commands/mail.py", line 427, in __send_message
    'patch': (__build_message, 'patch "%s"' % args[0]),
TypeError: not all arguments converted during string formatting
[cel@klimt linux]$ stg version
Stacked GIT 0.22-19-gca7d-dirty
git version 2.25.2
Python version 2.7.5 (default, Apr 11 2018, 17:41:36) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-28.0.1)]
[cel@klimt linux]$
```

According to https://stackoverflow.com/questions/18053500/typeerror-not-all-arguments-converted-during-string-formatting-python, the % operator in the following code is incorrect or deprecated:

```
363 def __send_message(type, tmpl, options, *args):
364     """Message sending dispatcher.
365     """
366     (build, outstr) = {
367         'cover': (__build_cover, 'the cover message'),
368         'patch': (__build_message, 'patch "%s"' % args[0]),
369     }[type]
```

When rewritten thusly:

```
363 def __send_message(type, tmpl, options, *args):
364     """Message sending dispatcher.
365     """
366     (build, outstr) = {
367         'cover': (__build_cover, 'the cover message'),
368         'patch': (__build_message, 'patch "{0}"'.format(args[0])),
369     }[type]
```

Then "stg mail -a -e" works as expected.

**Comment by jpgrayson** [comment]
Thank you for reporting this issue, @chucklever.

I was able to reproduce and repair the problem. Although I took a different approach than you suggested. Instead of using `str.format()` instead of interpolation (the `%` operator), I did a little bit of restructuring in the offending `__send_message()` function such that we are no longer exposed to `*args` for cover letters being used in the context of individual patches.

N.B. the minimum repair would be to wrap `args[0]` as a 1-tuple, i.e.:

``` python
'patch': (__build_message, 'patch "%s"' % (args[0], )),
```
This is actually the only correct/sure-fire way to avoid this kind of pathology when using a single variable with the interpolation operator.
